src/Appwrite/Platform/Workers/Audits.php

```diff
@@ -63,23 +80,58 @@ public function action(Message $message, Database $dbForProject): void
         $userEmail = $user->getAttribute('email', '');
         $userType = $user->getAttribute('type', Auth::ACTIVITY_TYPE_USER);
 
-        $audit = new Audit($dbForProject);
-        $audit->log(
-            userId: $user->getInternalId(),
-            // Pass first, most verbose event pattern
-            event: $event,
-            resource: $resource,
-            userAgent: $userAgent,
-            ip: $ip,
-            location: '',
-            data: [
+        // Create event data
+        $eventData = [
+            'userId' => $user->getInternalId(),
+            'event' => $event,
+            'resource' => $resource,
+            'userAgent' => $userAgent,
+            'ip' => $ip,
+            'location' => '',
+            'data' => [
                 'userId' => $user->getId(),
                 'userName' => $userName,
                 'userEmail' => $userEmail,
                 'userType' => $userType,
                 'mode' => $mode,
                 'data' => $auditPayload,
-            ]
-        );
+            ],
+            'timestamp' => DateTime::formatTz(DateTime::now())
+        ];
+
+        self::$logs[] = $eventData;
+
+        // Check if we should process the batch by checking both for the batch size and the elapsed time
+        $batchSize = $this->getBatchSize();
+        $shouldProcessBatch = count(self::$logs) >= $batchSize;
+        if (!$shouldProcessBatch && count(self::$logs) > 0) {
+            $oldestEventTime = self::$logs[0]['timestamp'];
+            $shouldProcessBatch = (time() - $oldestEventTime) >= self::BATCH_AGGREGATION_INTERVAL;
+        }
+
+        if ($shouldProcessBatch) {
+            Console::log('Processing batch with ' . count(self::$logs) . ' events');
+
+            $audit = new Audit($dbForProject);
+            $batchEvents = array_map(function ($event) {
+                return [
+                    'userId' => $event['userId'],
+                    'event' => $event['event'],
+                    'resource' => $event['resource'],
+                    'userAgent' => $event['userAgent'],
+                    'ip' => $event['ip'],
+                    'location' => $event['location'],
+                    'data' => $event['data'],
+                    'timestamp' => $event['timestamp']
+                ];
+            }, self::$logs);
+
+            $audit->logBatch($batchEvents);
+
+            // Clear the pending events after successful batch processing
+            self::$logs = [];
```

Comment:
If the log call fails, we won't clear the batch, is that expected?

Reply:
Im not sure whats the best approach here. should we hold onto the jobs in case of failure and try them again in the next batch? 🤔  Or just clear them upon first sign of failure?

Reply:
I think it's better to clear on error here:
- We'll still have all the data in the failed queue
- Not retrying automatically is consistent with other workers
- We avoid potential for filling memory constantly in the case where we have consistent failures